website/src/documentation/programming-guide.md

```diff
@@ -1970,7 +1976,1076 @@ records.apply("WriteToText",
+**shippingAddress.deliveryNotes** fields, and a false value filled in for the **transactions.isFlagged** field.
+
+##### **Removing fields from a schema**
+`DropFields` allows specific fields to be dropped from a schema. Input rows will have their schemas truncated, and any 
+values for dropped fields will be removed from the output. Nested fields can also be dropped using the field selection 
+syntax.
+
+For example, the following snippet
+
+```java
+purchases.apply(DropFields.fields(“userId”, “shippingAddress.streetAddress”));
+```
+
+Results in a copy of the input with those two fields and their corresponding values removed.
+
+##### **Renaming schema fields**
+`RenameFields` allows specific fields in a schema to be renamed. The field values in input rows are left unchanged, only 
+the schema is modified. This transform is often used to prepare records for output to a schema-aware sink, such as an 
+RDBMS, to make sure that the `PCollection` schema field names match that of the output. It can also be used to rename
+fields generated by other transforms to make them more usable (similar to SELECT AS in SQL). Nested fields can also be
+renamed using the field-selection syntax.
+
+For example, the following snippet
+
+```java
+purchases.apply(RenameFields.<PurchasePojo>create()
+  .rename(“userId”, “userIdentifier”)
+  .rename(“shippingAddress.streetAddress”, “shippingAddress.street”));
+```
+
+Results in the same set of unmodified input elements, however the schema on the PCollection has been changed to rename 
+**userId** to **userIdentifier** and **shippingAddress.streetAddress** to **shippingAddress.street**.
+
+##### **Converting between types**
+As mentioned, Beam can automatically convert between different Java types, as long as those types have equivalent
+schemas. One way to do this is by using the `Convert` transform, as follows.
+
+```java
+PCollection<PurchaseBean> purchaseBeans = readPurchasesAsBeans();
+PCollection<PurchasePojo> pojoPurchases = 
+    purchaseBeans.apply(Convert.to(PurchasePojo.class));
+```
+
+Beam will validate that the inferred schema for `PurchasePojo` matches that of the input `PCollection`, and will
+then cast to a `PCollection<PurchasePojo>`.
+
+Since the `Row` class can support any schema, any `PCollection` with schema can be cast to a `PCollection` of rows, as
+follows.
+
+```java
+PCollection<Row> purchaseRows = purchaseBeans.apply(Convert.toRows());
+```
+
+If the source type is a single-field schema, Convert will also convert to the type of the field if asked, effectively
+unboxing the row. For example, give a schema with a single INT64 field, the following will convert it to a
+`PCollection<Long>`
+
+```java
+PCollection<Long> longs = rows.apply(Convert.to(TypeDescriptors.longs()));
+```
+
+In all cases, type checking is done at pipeline graph construction, and if the types do not match the schema then the
+pipeline will fail to launch.
```

Comment:
What about adding an example of interop with `SqlTransform` mid-pipeline? Or do we not want to encourage that?

Reply:
I think it's a fine example, but let's hold off. we will need a followup PR that adds a bunch of examples anyway